src/Button.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { InternalLink } from "./Link";
+import { Button as MaterialButton } from "@material-ui/core";
+import { ButtonProps } from "@material-ui/core/Button/Button";
+
+interface Props {
+  /** Supports both internal and external links */
+  to?: string;
+  component?: any;
+}
+
+/** Use this component instead of MaterialUI's Button or ReactRouter's Link */
+export const Button: React.FC<Props & ButtonProps> = props => {
```

Comment:
@ry What do you think of this? This just makes sure you don't use href by accident.
```suggestion
export const Button: React.FC<Props & Omit<ButtonProps, 'href'> = props => {
```
If this is changed make sure to change it [here](https://github.com/denoland/deno_website2/pull/91/files#diff-01e857eda18d514416246bb1ff1e7510R16) too.

Reply:
Why can't we just use `<a href>`... I don't understand why getting a link is requiring so much infrastructure.

Reply:
@ry https://github.com/denoland/deno_website2/issues/86#issuecomment-547166068

Reply:
Thank you @lucacasonato - I missed that when you originally posted it. That was well put and very much helps me justify this.


@josephsintum - could you please copy @lucacasonato's comment into the top of src/Link.ts ? (since it is not at all obvious why this is necessary)

Reply:
I don't understand why react-router's Link doesn't understand the difference between external and internal links tho.

Reply:
React doesn't handle routing 
So if `<a href>` is used, react will render the anchor tag without the SPA routing logic, then the whole page is reloaded which is not a great experience
so we use external packages like react-router which provides a `<Link to="...">` to handle routing, 
The link need to be styled by material ui, so react-router <Link> is wrapped with material ui <Link>

Reply:
> I don't understand why react-router's Link doesn't understand the difference between external and internal links tho.

I agree - that is stupid, they should just handle external links as external links.